src/event-mixin.js

```diff
@@ -226,7 +258,28 @@ export function addEventListener(type, fn, optionsOrCapture) {
     this.__eventListenerCount = 0;
   }
   this.__eventListenerCount++;
-  let wrappedFn = function(e) {
+
+  // The function might be used for multiple events, so we need to keep track of
+  // the event type, options, and node into this object where the key determines
+  // the Id of the wrapped function (type+options), and the value is a WeakMap where
+  // we associate a node to the wrapped function.
+  fn.__eventWrappers = fn.__eventWrappers || {};
+  // This event listener might be added multiple times, we need to be able to remove
+  // all the wrappers we add.
+  const eventWrapperId = getEventWrapperId(type, optionsOrCapture);
+  const wrappersForType = fn.__eventWrappers[eventWrapperId] || new WeakMap();
```

Comment:
Have a performance concern here. Almost always there will be a new wrapper id when addEventListener is called and this means we'll basically make a new WeakMap for every call to addEventListener. It seems highly likely this will result in poor performance. At the least we must test this and I think it's probably better to refactor this to reduce the number of WeakMaps needed.